src/components/InputContainer/InputContainer.js

```diff
@@ -70,12 +65,13 @@ class InputContainer extends React.Component {
       label,
       id,
       positionLabelUpper,
+      inputRef,
       ...props
     } = this.props
 
     return (
       <div className={sparkClassName('utility', 'JavaScript')}>
-        <div className={this.className} {...props}>
+        <div className={this.className} ref={inputRef} {...props}>
```

Comment:
not sure what the `ref` is providing here?

Reply:
in spark 9 the bindUIEvents functions changed, it needs to be in a wrapper component that has the input as the child. here's an example:
`const bindUIEvents = (element) => {
  const field = element.querySelector('input');
  field.addEventListener('input', () => {
    runValidation(element);
  });
};`